comptoir-ejb/src/main/java/be/valuya/comptoir/service/SaleService.java

```diff
@@ -335,6 +337,16 @@ public void cancelOpenSale(Sale sale) {
             accountService.removeAccountingEntry(accountingEntry);
         }
 
+        // remove customer loyalty entry
+        try {
+            CustomerLoyaltyAccountingEntry saleCustomerLoyaltyEntry = findSaleCustomerLoyaltyEntry(sale);
+            if (saleCustomerLoyaltyEntry != null) {
+                removeCustomerLoyaltyAccountingEntry(saleCustomerLoyaltyEntry);
+            }
+        } catch (IllegalStateException e) {
```

Comment:
Whenever we catch something like this, we can now the Exception type isn't appropriate.